src/libraries/Directory.Build.props

```diff
@@ -323,6 +323,9 @@
 
   <ItemGroup Condition="'$(IsTestProject)' == 'true'">
     <ProjectReference Include="$(CommonTestPath)TestUtilities\TestUtilities.csproj" />
+    
+    <ProjectReference Include="$(CommonTestPath)AppleTestRunner\AppleTestRunner.csproj" Condition="'$(TargetOS)' == 'iOS' or '$(TargetOS)' == 'tvOS'" />
+    <ProjectReference Include="$(CommonTestPath)AndroidTestRunner\AndroidTestRunner.csproj" Condition="'$(TargetOS)' == 'Android'" />
```

Comment:
This looks weird. Why would a test project have a reference to its test runner? We don't have that for non mobile OSs.

Reply:
> This looks weird. Why would a test project have a reference to its test runner? We don't have that for non mobile OSs.

For non mobile OSs the test runner is XUnit and we get it via a NuGet package. 

In this case the Main application test runner are these test runners which need to wrap xunit in order to make it a runnable mobile app. We need the runners to be in the publish directory and be part of the bundle. Before this change we where hard coding/calculating the path for the runner and copying it manually to the publish dir. Making various things difficult, these were built during Mono builds, which mean that if you follow the split build model like in CI you need to carry them over as artifacts and make sure they land on the right location when building the libraries tests and flow the right runtime configuration, etc. Also in order to reduce size in the bundles we want to run the linker to trim the app, which relies on the SDK list of files that are part of the closure, since we were copying manually these where not consider causing the trim result to be wrong. Another thing, incremental inner dev loop changing these projects was not easy as  you needed to go build them on mono and actually use the same configuration you used for the mono runtime for these to be found, etc.

This is something that all test projects use in mobile so the most natural thing to solve those problems was adding a P2P.

Reply:
I would have really appreciated if we could have first have talked about this in an issue before submitting a PR and merging it without letting me review the design.

I have couple of comments:
1. You are adding a compiler reference from the test app to the test runner. There should never be such a reference as it breaks separation of concerns. I recommend to add the `ReferenceOutputAssembly=false` attributes to the project references to prevent that.
2. Adding a P2P contract between the test app and the test runner means that you will always restore and build the test runner as well, either incrementally or full which adds significant cost. I don't understand why this is necessary.
3. Why can't the test runners be built and trimmed in pretest.proj and their output added to the `ResolvedFileToPublish` item for the test apps?